src/lib.rs

```diff
@@ -118,6 +121,237 @@ pub fn builder() -> Builder {
+                .multiple(true),
+        ]) // .args()
+        .get_matches_from(bindgen_args);
+
+    let mut builder = builder();
+
+    if let Some(header) = matches.value_of("header") {
+        builder = builder.header(header);
+    } else {
+        return Err(Error::new(ErrorKind::Other, "Header not found"));
+    }
+
+    if let Some(hidden_types) = matches.values_of("blacklist-type") {
+        for ty in hidden_types {
+            builder = builder.hide_type(ty);
+        }
+    }
+
+    if matches.is_present("builtins") {
+        builder = builder.emit_builtins();
+    }
+
+    if let Some(prefix) = matches.value_of("ctypes-prefix") {
+        builder = builder.ctypes_prefix(prefix);
+    }
+
+    if let Some(dummy) = matches.value_of("dummy-uses") {
+        builder = builder.dummy_uses(dummy);
+    }
+
+    if let Some(links) = matches.values_of("dynamic") {
+        for library in links {
+            builder = builder.link(library);
+        }
+    }
+
+    if matches.is_present("emit-clang-ast") {
+        builder = builder.emit_clang_ast();
+    }
+
+    if matches.is_present("enable-cxx-namespaces") {
+        builder = builder.enable_cxx_namespaces();
+    }
+
+    if let Some(links) = matches.values_of("framework") {
+        for framework in links {
+            builder = builder.link_framework(framework);
+        }
+    }
+
+    if matches.is_present("ignore-functions") {
+        builder = builder.ignore_functions();
+    }
+
+    if matches.is_present("ignore-methods") {
+        builder = builder.ignore_methods();
+    }
+
+    if matches.is_present("no-unstable-rust") {
+        builder = builder.no_unstable_rust();
+    }
+
+    if let Some(opaque_types) = matches.values_of("opaque-type") {
+        for ty in opaque_types {
+            builder = builder.opaque_type(ty);
+        }
+    }
+
+    if let Some(lines) = matches.values_of("raw-line") {
+        for line in lines {
+            builder = builder.raw_line(line);
+        }
+    }
+
+    if let Some(links) = matches.values_of("static") {
+        for library in links {
+            builder = builder.link_static(library);
+        }
+    }
+
+    if matches.is_present("use-core") {
+        builder = builder.use_core();
+    }
+
+    if let Some(whitelist) = matches.values_of("whitelist-function") {
+        for regex in whitelist {
+            builder = builder.whitelisted_function(regex);
+        }
+    }
+
+    if let Some(whitelist) = matches.values_of("whitelist-type") {
+        for regex in whitelist {
+            builder = builder.whitelisted_type(regex);
+        }
+    }
+
+    if let Some(whitelist) = matches.values_of("whitelist-var") {
+        for regex in whitelist {
+            builder = builder.whitelisted_var(regex);
+        }
+    }
+
+    for arg in clang_args.into_iter().skip(1) {
+        builder = builder.clang_arg(arg);
+    }
+
+    // FIXME: hrrrmmm
+    let _output = if let Some(path) = matches.value_of("output") {
```

Comment:
Since this is a helper, I guess this can return `Result<Builder, Option<Box<io::Write>>` (I'd rather decide what to do with the missing output case in the caller).